src/openzeppelin/token/erc20.cairo

```diff
@@ -0,0 +1,200 @@
+trait IERC20 {
+    fn name() -> felt;
+    fn symbol() -> felt;
+    fn decimals() -> u8;
+    fn total_supply() -> u256;
+    fn balance_of(account: ContractAddress) -> u256;
+    fn allowance(owner: ContractAddress, spender: ContractAddress) -> u256;
+    fn transfer(recipient: ContractAddress, amount: u256) -> bool;
+    fn transfer_from(sender: ContractAddress, recipient: ContractAddress, amount: u256) -> bool;
+    fn approve(spender: ContractAddress, amount: u256) -> bool;
+}
+
+#[contract]
+mod ERC20 {
+    use openzeppelin::token::erc20::IERC20;
+    use starknet::get_caller_address;
+    use starknet::contract_address_const;
+    use starknet::ContractAddressZeroable;
+    use zeroable::Zeroable;
+
+    struct Storage {
+        _name: felt,
+        _symbol: felt,
+        _total_supply: u256,
+        _balances: LegacyMap::<ContractAddress, u256>,
+        _allowances: LegacyMap::<(ContractAddress, ContractAddress), u256>,
+    }
+
+    #[event]
+    fn Transfer(from: ContractAddress, to: ContractAddress, value: u256) {}
+
+    #[event]
+    fn Approval(owner: ContractAddress, spender: ContractAddress, value: u256) {}
+
+    impl ERC20 of IERC20 {
+        fn name() -> felt {
+            _name::read()
+        }
+
+        fn symbol() -> felt {
+            _symbol::read()
+        }
+
+        fn decimals() -> u8 {
+            18_u8
+        }
+
+        fn total_supply() -> u256 {
+            _total_supply::read()
+        }
+
+        fn balance_of(account: ContractAddress) -> u256 {
+            _balances::read(account)
+        }
+
+        fn allowance(owner: ContractAddress, spender: ContractAddress) -> u256 {
+            _allowances::read((owner, spender))
+        }
+
+        fn transfer(recipient: ContractAddress, amount: u256) -> bool {
+            let sender = get_caller_address();
+            _transfer(sender, recipient, amount);
+            true
+        }
+
+        fn transfer_from(sender: ContractAddress, recipient: ContractAddress, amount: u256) -> bool {
+            let caller = get_caller_address();
+            _spend_allowance(sender, caller, amount);
+            _transfer(sender, recipient, amount);
+            true
+        }
+
+        fn approve(spender: ContractAddress, amount: u256) -> bool {
+            let caller = get_caller_address();
+            _approve(caller, spender, amount);
+            true
+        }
+    }
+
+    #[constructor]
+    fn constructor(name: felt, symbol: felt, initial_supply: u256, recipient: ContractAddress) {
+        initializer(name, symbol, initial_supply, recipient);
+    }
```

Comment:
In understand that we may want the constructor to mint an initial supply, but I'm not sure that should be part of the `initializer`

```suggestion
    fn constructor(name: felt, symbol: felt, initial_supply: u256, recipient: ContractAddress) {
        initializer(name, symbol);
        _mint(recipient, initial_supply);
    }
```

Reply:
Agreed, good call